### ISSUE TITLE
does pre-assembly require the assembly script to work?

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/pre-assembly#87]
is that necessary as a co-installed application to run remediation?

**Comment by atz** [comment]
Based on [these lines](https://github.com/sul-dlss/pre-assembly/blob/master/lib/pre_assembly/remediation/remediate.rb#L150-L152) in `remediate.rb`, assembly is required in a hardcoded location, accessible to the same user:
```
def run_assembly_robot(name)
  `BUNDLE_GEMFILE=~/assembly/current/Gemfile ROBOT_ENVIRONMENT=#{ENV['ROBOT_ENVIRONMENT']} bundle exec ~/assembly/current/bin/run_robot dor:assemblyWF:#{name} -e #{ENV['ROBOT_ENVIRONMENT']} -d #{@pid}`
end
```

Suffice to say, this is a really poorly defined interface between two applications.  Our options include:
- allow `remediate.rb` to break in our new servers
  - maintain "old" servers in parallel with new ones indefinitely, specifically to provide this functionality, OR
  - allow "old" servers to be decommissioned, effectively killing this functionality
- install/deploy fully functional `assembly` app as part of new servers to satisfy this dependency
- have `assembly` consume all the functionality of `pre-assembly` because `pre-assembly` is not sufficiently distinct or _cannot_ be disentangled
- develop proper interface in `assembly` for `pre-assembly` to invoke and convert `remediate.rb` to use it
- extract necessary functionality from `assembly` and reproduce it in `pre-assembly`

None of these are really attractive.

**Comment by atz** [comment]
We resolved this in #92, meaning assembly is no longer required for pre-assembly.
